lib/ruby_lsp/responses/response.rb

```diff
@@ -0,0 +1,18 @@
+# typed: strict
+# frozen_string_literal: true
+
+module RubyLsp
+  module Responses
+    class Response
+      extend T::Sig
+      extend T::Generic
+
+      abstract!
+
+      ResponseType = type_member
+
+      sig { abstract.returns(ResponseType) }
+      def result; end
```

Comment:
Maybe the names should be these instead?

- `ResponseBuilders`
- `ResponseBuilder`
- `ResponseBuilder#response`